Feature/k8s

cableman opened the pull request.

cableman committed:
DDBTEAM-553: Added image build and helm charts

cableman committed:
DDBTEAM-553: Added helm chart lint

kasperg reviewed (commented):
Remember to remove parts of the old deployment setup.

This includes:

- [Github workflows for deployment](https://github.com/danskernesdigitalebibliotek/ddb-follow-searches/blob/feature/k8s/.github/workflows/push.yml#L92-L120)
- [Parts](https://github.com/danskernesdigitalebibliotek/ddb-follow-searches/blob/master/Envoy.blade.php) [of](https://github.com/danskernesdigitalebibliotek/ddb-follow-searches/blob/master/composer.json#L35) the old deployment setup using Envoy.
- [Documentation referring to the old deployment setup](https://github.com/danskernesdigitalebibliotek/ddb-follow-searches#server-setup)

cableman committed:
DDBTEAM-561: Added docker compose setup

cableman committed:
DDBTEAM-561: Trying to add SSL support for database connection

cableman committed:
DDBTEAM-561: Updated helm chart

cableman committed:
DDBTEAM-561: Updated readme with helm information

cableman committed:
DDBTEAM-570: Added network policies to chart

cableman committed:
DDBTEAM-570: Updated readme with namespaces

cableman committed:
DDBTEAM-591: Fixed policies and cors in k8s

cableman committed:
DDBTEAM-591: Renamed policy

kasperg reviewed (changes requested):
My comments regarding danskernesdigitalebibliotek/ddb-material-list#1 also apply here.

Also see my previous comment about removing [parts](https://github.com/danskernesdigitalebibliotek/ddb-follow-searches/blob/master/Envoy.blade.php) [of](https://github.com/danskernesdigitalebibliotek/ddb-follow-searches/blob/master/composer.json#L35) the old deployment setup using Envoy.

kasperg committed:
Lock Dredd at version 12

For some reason our tests start failing from version 12 and on.

We have implement the same lock for the material list service.

kasperg committed:
Limit Composer to version 1.x

Setup-php will install the latest version of Composer by default and
version 2 has just been released. Limit to version 1.x for now to
keep the continuous integration process running.

kasperg commented:
The specification test failure is not related to this PR. It will be fixed when #4 is merged.

cableman committed:
DDBTEAM-562: Updated based on code review

cableman commented:
Updated the code with changes from this review and the one for material list.

I regard to en Envoy it seams to be an integrated part of the Larvel and as I don't known anything about Larvel I think you should make another PR removing it. It do not seam to block anything here.

kasperg commented on .github/workflows/pull_request.yml:
```diff
@@ -44,3 +46,14 @@ jobs:
+      - uses: azure/setup-helm@v1
+        name: Install helm
+        with:
+          version: v3.3.4
```
Same quibbles are for material list regarding versioning.

kasperg committed:
Remove Envoy

Kubernetes is replacing our current hosting platform and deployment
workflow.

kasperg reviewed (commented):
> I regard to en Envoy it seams to be an integrated part of the Larvel and as I don't known anything about Larvel I think you should make another PR removing it. It do not seam to block anything here.

It is a Composer package and a configuation file that needs to be removed. I linked to the parts in my comment. I have removed them it a commit added to the PR in a25939d.
